src/lib.rs

```diff
@@ -92,32 +95,32 @@ fn get_compact_symbol_table_impl(
     binary_data: &[u8],
     debug_data: &[u8],
     breakpad_id: &str,
-) -> Option<compact_symbol_table::CompactSymbolTable> {
+) -> Result<compact_symbol_table::CompactSymbolTable> {
     let mut reader = Cursor::new(binary_data);
-    if let Ok(hint) = goblin::peek(&mut reader) {
-        match hint {
-            Hint::Elf(_) => {
-                return elf::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::Mach(_) => {
-                return macho::get_compact_symbol_table(binary_data, breakpad_id);
-            }
-            Hint::MachFat(_) => {
-                let multi_arch = mach::MultiArch::new(binary_data).ok()?;
-                for fat_arch in multi_arch.iter_arches().filter_map(Result::ok) {
-                    let arch_slice = fat_arch.slice(binary_data);
-                    if let Some(table) = macho::get_compact_symbol_table(arch_slice, breakpad_id) {
-                        return Some(table);
-                    }
+    match goblin::peek(&mut reader)? {
+        Hint::Elf(_) => elf::get_compact_symbol_table(binary_data, breakpad_id),
+        Hint::Mach(_) => macho::get_compact_symbol_table(binary_data, breakpad_id),
+        Hint::MachFat(_) => {
+            let multi_arch = mach::MultiArch::new(binary_data).ok();
+            for fat_arch in multi_arch
+                .unwrap()
+                .iter_arches()
+                .filter_map(std::result::Result::ok)
+            {
+                let arch_slice = fat_arch.slice(binary_data);
+                if let Ok(table) = macho::get_compact_symbol_table(arch_slice, breakpad_id) {
+                    return Ok(table);
                 }
             }
-            Hint::PE => {
-                return pdb::get_compact_symbol_table(debug_data, breakpad_id);
-            }
-            _ => {}
+            Err(GetSymbolsError::InvalidInputError(
+                "Incompatible system architecture",
+            ))
```

Comment:
I think we can do a bit better for this error. Because we're iterating over multiple sub-binaries here, and discard the errors from failing calls to `macho::get_compact_symbol_table`, `InvalidInputError` might not be entirely accurate.
How about this: For the first call to `macho::get_compact_symbol_table`, if that call returns an error, store that error in a `first_error: Option<GetSymbolsError>` variable. Then, if we get to the end of the loop (meaning all sub-binaries failed in some way), and `first_error` is not `None`, return that error. Otherwise return `InvalidInputError` as you're doing now.